### ISSUE TITLE
Crypto.Cipher.ARC2 does not accept effective_keylen=40 

### ISSUE THREAD

**Issue by fishilico** [Legrandin/pycryptodome#267]
Hello,
In order to decrypt some data in a PKCS#12 keystore encrypted using ``pbeWithSHA1And40BitRC2-CBC`` (OID 1.2.840.113549.1.12.1.6), ARC2 needs to be used with a 40-bit key (which is derived from a password). I am therefore trying to use:
```py
Crypto.Cipher.ARC2.new(key, Crypto.Cipher.ARC2.MODE_CBC, iv=iv, effective_keylen=40)
```

This triggers an exception in https://github.com/Legrandin/pycryptodome/blob/v3.7.3/lib/Crypto/Cipher/ARC2.py#L76:
```
  File "/usr/lib/python3.7/site-packages/Crypto/Cipher/ARC2.py", line 78, in _create_base_cipher
    "(not %d)" % effective_keylen)
ValueError: 'effective_key_len' must be no larger than 1024 (not 40)
```

This is because the condition is currently ``if not (40 < effective_keylen <= 1024):`` instead of ``... 40 <= effective_keylen ...``. The code in C accepts the value 40, according to https://github.com/Legrandin/pycryptodome/blob/v3.7.3/src/ARC2.c#L80:
```c
if ((effective_key_bits < 40) || (effective_key_bits > 1024))
    return ERR_KEY_SIZE;
```

Could you please make ``Crypto.Cipher.ARC2.new`` accept ``effective_keylen=40`` by using ``if not (40 <= effective_keylen <= 1024):`` in ``lib/Crypto/Cipher/ARC2.py``?

By the way, even though using 40-bit keys with ARC2 may sound like a weak algorithm in 2019, it is still the default encryption algorithm which is used to encrypt certificates generated with ``keytool -genkeypair -keyalg rsa -sigalg SHA256withRSA -storetype pkcs12 -keystore store.jks`` with OpenJDK Java 10.

**Comment by Legrandin** [comment]
Sure, it makes sense and I just fixed it with 36c0bcb85162a7e18b7cff4617372acd0b41fa0e.
However, it will take a few days if not weeks to make it into a release...

**Comment by fishilico** [comment]
Thanks for your quick reply! No problem for the delay: I can wait for the next release before migrating my code away from the custom ARC2 implementation I am currently using.
